### PR TITLE
feat(notifications): new notification settings home for V2 settings

### PR DESCRIPTION
**Pull request by scefali**
This PR updates the notification settings home page for the V2 of notification settings using the new models

Screen Shot 2023-08-25 at 11 48 23 AM

### CODE REVIEW

File: static/app/views/settings/account/notifications/notificationSettingsV2.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      </FieldWrapper>
+    );
+  };
+
+  const legacyFields = SELF_NOTIFICATION_SETTINGS_TYPES.map(
+    type => NOTIFICATION_SETTING_FIELDS[type] as FieldObject
+  );
+
+  // use 0 as stale time because we change the values elsewhere
+  const {data: initialLegacyData, isLoading} = useApiQuery<{[key: string]: string}>(
```
Comment (scttcper) [inline static/app/views/settings/account/notifications/notificationSettingsV2.tsx:71]: you'll need to handle errors somehow

Comment (scefali) [reply static/app/views/settings/account/notifications/notificationSettingsV2.tsx:71]: @scttcper not sure what we'd do in an error scenario, any suggestions?

**Review by scttcper — APPROVED** [review]

File: static/app/views/settings/account/notifications/notificationSettingsV2.tsx [hunk 2]
```diff
@@ -68,12 +69,17 @@ function NotificationSettings({organizations}: Props) {
...
-    ['/users/me/notifications/'],
-    {
-      staleTime: 0,
-    }
-  );
+  const {
+    data: initialLegacyData,
+    isLoading,
+    isError,
+  } = useApiQuery<{[key: string]: string}>(['/users/me/notificationsddd/'], {
```
Comment (scttcper) [inline static/app/views/settings/account/notifications/notificationSettingsV2.tsx:76]:
```suggestion
  } = useApiQuery<{[key: string]: string}>(['/users/me/notifications/'], {
```

Comment (scefali) [reply static/app/views/settings/account/notifications/notificationSettingsV2.tsx]: @scttcper lol yea good catch

**Pull request merged**